lib/kontena/machine/digital_ocean/cloudinit.yml

```diff
@@ -48,6 +52,12 @@ coreos:
         Name=weave*
 
         [Network]
+    - name: zz-default.network
+      runtime: false
+      content: |
+        # default should only match real network interfaces
+        [Match]
```

Comment:
If you're overriding an existing unit, you need to have more than just the `[Match]` section, this is missing the `[Network]` section.

You want something like

```
[Match]
Name=eth*

[Network]
DHCP=yes

[DHCP]
UseMTU=true
UseDomains=true
```

I suppose the reason this works for the initial boot is that the cloud-config is applied after systemd-networkd has been started with the original default `.network` config. On a reboot, the `/etc/systemd/network` config will apply, and the default network connectivity will be broken due to a lack of DHCP.